From b1065713b18ddfe1649c626798936cf5b4687007 Mon Sep 17 00:00:00 2001
From: "ashritha.udpikar" <ashritha.udpikar@lge.com>
Date: Tue, 12 Sep 2023 14:19:39 +0530
Subject: [PATCH] Fixed crash issue of TTS-Service

:Release Notes:
Fixed crash issue of TTS-Service

:Detailed Notes:
[RDX_CRASH][webos] /usr/sbin/tts-service
                   in __poll (libc.so.6 + 0xde830)
Program terminated with signal SIGSEGV, Segmentation fault.

:Testing Performed:
Locally Tested.

:QA Notes:

:Issues Addressed:
[WRP-26215][RDX_CRASH][webos] /usr/sbin/tts-service
           in __poll (libc.so.6 + 0xde830)
---

Upstream-Status: Pending

 src/core/lib/surface/call.cc | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/core/lib/surface/call.cc b/src/core/lib/surface/call.cc
index dab2f50c403..53a9a3ac2b0 100644
--- a/src/core/lib/surface/call.cc
+++ b/src/core/lib/surface/call.cc
@@ -900,6 +900,11 @@ class PublishToAppEncoder {
   }
 
   void Append(grpc_slice key, grpc_slice value) {
+    gpr_log(GPR_INFO, "Count: %d", dest_->count);
+    gpr_log(GPR_INFO, "Capacity is: %d", dest_->capacity);
+    if (dest_->count == dest_->capacity) {
+      gpr_log(GPR_ERROR, "Too many metadata entries.");
+    }
     auto* mdusr = &dest_->metadata[dest_->count++];
     mdusr->key = key;
     mdusr->value = value;
@@ -919,7 +924,7 @@ void FilterStackCall::PublishAppMetadata(grpc_metadata_batch* b,
   dest = buffered_metadata_[is_trailing];
   if (dest->count + b->count() > dest->capacity) {
     dest->capacity =
-        std::max(dest->capacity + b->count(), dest->capacity * 3 / 2);
+        std::max(dest->capacity + b->count(), dest->capacity * 4 / 2);
     dest->metadata = static_cast<grpc_metadata*>(
         gpr_realloc(dest->metadata, sizeof(grpc_metadata) * dest->capacity));
   }
-- 
2.17.1

